src/Telemetry/OpenTelemetryScope.cs

```diff
@@ -0,0 +1,191 @@
+using System.ClientModel;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using OllamaSharp.Models.Chat;
+
+using static OllamaSharp.Telemetry.OpenTelemetryConstants;
+
+namespace OllamaSharp.Telemetry;
+
+internal class OpenTelemetryScope : IDisposable
+{
+	private static readonly ActivitySource _chatSource = new("OllamaSharp.ChatClient");
+	private static readonly Meter _chatMeter = new("OllamaSharp.ChatClient");
+
+	// TODO: add explicit histogram buckets once System.Diagnostics.DiagnosticSource 9.0 is used
+	private static readonly Histogram<double> _duration = _chatMeter.CreateHistogram<double>(GEN_AI_CLIENT_OPERATION_DURATION_METRIC_NAME, "s", "Measures GenAI operation duration.");

```

Comment:
You may want to create suitable buckets for the histogram based on what a typical duration would be, using exponential-ish bucket sizes. Eg https://github.com/dotnet/aspnetcore/blob/cfcb73ab7e63ad193d20566efb23b8bf93e494ba/src/Hosting/Hosting/src/Internal/HostingMetrics.cs#L30

Reply:
Given that you could have a model that's only a few hundred parameters, or one that is several hundreds of billions, I'm not sure that it's really possible to determine what the typical duration would be.

Reply:
@samsp-msft is is on .NET 10 only? I want to use it too

@aaronpowell users can customize buckets, the advice is part of the spec https://github.com/open-telemetry/semantic-conventions/blob/main/docs/gen-ai/gen-ai-metrics.md#metric-gen_aiclientoperationduration - without it everyone needs to customize all the time

Reply:
@aaronpowell - The spec seems to have good exponential buckets from 100ms to 80s. You should be setting those as the default buckets using the Advise option.
@lmolkova - that code is present in the .NET 9 branch so you shouldn't have to wait for .NET 10.

Reply:
OllamaSharp supports netstandard so the implementation will be restricted to that.

Reply:
do you build just one version, or can you have version specific ifdef's?